docs/architecture/adr-060-go-api-stability.md

```diff
@@ -0,0 +1,193 @@
+
+The following is the minimum set of public APIs that will be included in 1.0, in some form:
+
+- `abci`
+- `node` and related packages (e.g. possibly `config`, `libs/log`, and `version`)
+- Client APIs, i.e. `rpc/client`, `light`, and `privval`.
+- `crypto` (possibly as a separate repo)
+
+We may offer additional APIs as well, following further discussions internally and with other stakeholders. However, public APIs for providing custom components (e.g. reactors and mempools) are not planned for 1.0, but may be added in a later 1.x version if this is something we want to offer.
+
+For comparison, the following are the number of Tendermint imports in the Cosmos SDK (excluding tests), which should be mostly satisfied by the planned APIs.
+
+```
+      1 github.com/tendermint/tendermint/abci/server
+     73 github.com/tendermint/tendermint/abci/types
+      2 github.com/tendermint/tendermint/cmd/tendermint/commands
+      7 github.com/tendermint/tendermint/config
+     68 github.com/tendermint/tendermint/crypto
+      1 github.com/tendermint/tendermint/crypto/armor
+     10 github.com/tendermint/tendermint/crypto/ed25519
+      2 github.com/tendermint/tendermint/crypto/encoding
+      3 github.com/tendermint/tendermint/crypto/merkle
+      3 github.com/tendermint/tendermint/crypto/sr25519
+      8 github.com/tendermint/tendermint/crypto/tmhash
+      1 github.com/tendermint/tendermint/crypto/xsalsa20symmetric
+     11 github.com/tendermint/tendermint/libs/bytes
+      2 github.com/tendermint/tendermint/libs/bytes.HexBytes
+     15 github.com/tendermint/tendermint/libs/cli
+      2 github.com/tendermint/tendermint/libs/cli/flags
+      2 github.com/tendermint/tendermint/libs/json
+     30 github.com/tendermint/tendermint/libs/log
+      1 github.com/tendermint/tendermint/libs/math
+     11 github.com/tendermint/tendermint/libs/os
+      4 github.com/tendermint/tendermint/libs/rand
+      1 github.com/tendermint/tendermint/libs/strings
+      5 github.com/tendermint/tendermint/light
+      1 github.com/tendermint/tendermint/mempool
+      3 github.com/tendermint/tendermint/node
+      5 github.com/tendermint/tendermint/p2p
+      4 github.com/tendermint/tendermint/privval
+     10 github.com/tendermint/tendermint/proto/tendermint/crypto
+      1 github.com/tendermint/tendermint/proto/tendermint/libs/bits
+     24 github.com/tendermint/tendermint/proto/tendermint/types
+      3 github.com/tendermint/tendermint/proto/tendermint/version
+      2 github.com/tendermint/tendermint/proxy
+      3 github.com/tendermint/tendermint/rpc/client
+      1 github.com/tendermint/tendermint/rpc/client/http
+      2 github.com/tendermint/tendermint/rpc/client/local
+      3 github.com/tendermint/tendermint/rpc/core/types
+      1 github.com/tendermint/tendermint/rpc/jsonrpc/server
+     33 github.com/tendermint/tendermint/types
+      2 github.com/tendermint/tendermint/types/time
+      1 github.com/tendermint/tendermint/version
```

Comment:
There's a lot in here not listed above, namely chunks of `libs` but especially `types`. Can probably fit `commands` in with `node` and co. Would be good to flesh this out a bit further, and especially to consider what busting up `types` is going to look like. https://github.com/tendermint/tendermint/issues/5379 is timely.

Reply:
Design of specific APIs was deliberately kept out-of-scope for this ADR, since it can get quite noisy and distracting. We'll have separate ADRs on this as we approach 1.0.

As for `libs`, we'll keep those internal - providing and supporting general-purpose utility APIs isn't really part of our product. If anything, this might belong in the SDK (since providing convenience APIs for app developers is what they do). `commands` should be split in such a way that the actual functionality is part of the public node API but the CLI user interface code is considered internal.